fedbiomed/common/channel_manager.py

```diff
@@ -0,0 +1,186 @@
+# This file is originally part of Fed-BioMed
+# SPDX-License-Identifier: Apache-2.0
+
+"""Interface with the node to node channel state database
+"""
+from typing import Union, List
+import base64
+
+from tinydb import TinyDB, Query
+
+from fedbiomed.common.utils import raise_for_version_compatibility, __default_version__
+from fedbiomed.common.constants import ErrorNumbers, __n2n_channel_element_version__
+from fedbiomed.common.db import DBTable
+from fedbiomed.common.exceptions import FedbiomedNodeToNodeError
+from fedbiomed.common.logger import logger
+
+
+_TableName = 'ChannelManager'
+
+
+class ChannelManager:
+    """Manage a node to node channel status database
+    """
+
+    def __init__(self, db_path: str):
+        """Constructor of the class
+
+        Args:
+            db_path: path to the n2n channel database
+
+        Raises:
+            FedbiomedNodeToNodeError: failed to access the database
+        """
+        try:
+            self._db = TinyDB(db_path)
+            self._db.table_class = DBTable
+        except Exception as e:
+            errmess = f'{ErrorNumbers.FB630.value}: failed to access the database with error: {e}'
+            logger.error(errmess)
+            raise FedbiomedNodeToNodeError(errmess)
```

Comment:
Thank you @srcansiz for the review.

3. Checking our current `CODING_RULES.md` you are right I should not be doing exception + logging => I will remove that.
1. 2. Regarding our coding rules again, we are free to catch or not system exceptions and re-raise as fedbiomed exceptions. So both implementing it or not are valid. For me it makes sense to catch it, because it indicates upper layer that it is a possible known failure case (vs an unexpected error). Also, it keeps coherence with current implementation of other DB access classes. I will implement the `FedbiomedError(...) from e` as you recommend.